library/objective-c/EnvoyEngineImpl.m

```diff
@@ -67,9 +67,12 @@ static envoy_filter_data_status ios_http_filter_on_request_data(envoy_data data,
 
   NSData *platformData = to_ios_data(data);
   NSArray *result = filter.onRequestData(platformData, end_stream);
+  // Result is typically a pair of status and entity, but uniquely in the case of
+  // ResumeIteration it will (optionally) contain additional pending elements.
+  envoy_headers *pending_headers = toNativeHeadersPtr(result.count == 3 ? result[2] : nil);
```

Comment:
```suggestion
  envoy_headers *pending_headers = toNativeHeadersPtr(result.count == 3 ? result[2] : NULL);
```

Reply:
This one should be `nil` - it's being passed as a missing objc object, not as a raw pointer.